src/js/api/auth/login.test.js

```diff
@@ -0,0 +1,64 @@
+import { login } from './login.js';
+
+global.fetch = jest.fn(() =>
+  Promise.resolve({
+    ok: true,
+    json: () =>
+      Promise.resolve({
+        accessToken: 'test',
+        name: 'test',
+        email: 'test@noroff.no',
+      }),
+    status: 'OK',
+  }),
+);
+
+global.localStorage = {
+  setItem: jest.fn(function (key, value) {
+    this[key] = value;
+  }),
+};
+
+describe('Login Function', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('Should login successfully and save token and profile in localstorage', async () => {
+    const email = 'testaccount4567@stud.noroff.no';
+    const password = 'Testaccount4567';
+
+    await login(email, password);
+
+    expect(fetch).toHaveBeenCalledWith(
+      expect.stringContaining('/social/auth/login'),
+      {
+        method: 'post',
+        body: JSON.stringify({ email, password }),
+        headers: expect.any(Object),
+      },
+    );
+
+    expect(localStorage.token).toEqual(JSON.stringify('test'));
+    expect(localStorage.profile).toEqual(
+      JSON.stringify({
+        name: 'test',
+        email: 'test@noroff.no',
+      }),
+    );
+  });
+
```

Comment:
Hi Marius,

Should there be a test to check if the token gets deleted after the login function is executed? I noticed that the login function deletes the token at some point. This is from the login.js file:

```
if (response.ok) {
  const profile = await response.json();
  save("token", profile.accessToken);
  delete profile.accessToken;  
  save("profile", profile);
  return profile;
}
```
It's only a suggestion, I am not sure if it is needed for this CA:)  Great work with the tests, Marius!

Best regards,

Alex